modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -65,23 +65,50 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 
 			if (refreshCaptcha && !hasEventAttached) {
 				hasEventAttached = true;
-				refreshCaptcha.addEventListener('click', () => {
-					var url = Liferay.Util.addParams(
-						't=' + Date.now(),
-						'<%= HtmlUtil.escapeJS(url) %>'
-					);
-
-					var captcha = document.getElementById(
-						'<portlet:namespace />captcha'
-					);
-
-					if (captcha) {
-						captcha.setAttribute('src', url);
-					}
-				});
+				refreshCaptcha.addEventListener('click', handleCaptcha);
 			}
 		}
 
+		function handleCaptcha() {
+			var captcha = document.getElementById(
+				'<portlet:namespace />captcha'
+			);
+
+			if(captcha){
+				if (<%= url.startsWith("/o/") %>) {
+					return getCaptchaSourceImage('<%= url %>', captcha);
+				}
+	
+				let url = Liferay.Util.addParams(
+					't=' + Date.now(),
+					'<%= HtmlUtil.escapeJS(url) %>'
+				);
+	
+				captcha.setAttribute('src', url);
+			}
+		}
+
+		function getCaptchaSourceImage(url, captcha) {
+			Liferay.Util.fetch(url, {
+				method: 'GET',
+			})
+			.then((response) => {
+				if (!response.ok) {
+					throw new Error();
+				}
+
+				return response.json();
+			})
+			.then((response) => {
+				captcha.setAttribute('src', response.image);
+			})
+			.catch((error) => {
+				console.error('An error occurred:', error);
+			});
+		}
+
+		handleCaptcha();
```

Comment:
Should make sure that the handleCaptcha will be called when the captcha is ready.

Reply:
Fixed it. Thanks